ingest/bulk_upload.py

```diff
@@ -35,6 +35,39 @@ def ICU_installed(es):
     return "analysis-icu" in [p["component"] for p in cc.plugins(h=None, format="json")]
 
 
+def wait_for_elasticsearch(host: str, wait: int) -> None:
+    """
+    Wait for elasticsearch to be available and healthy.
+
+    :param host: Host at which elasticsearch resides, for example
+    "http://localhost:9200".
+    :param wait: Number of seconds to wait before giving up and
+    throwing an exception.
+    :return: A non-exceptional return indicates that elasticsearch is healthy.
+    """
+    url = f"{host}/_cluster/health?wait_for_status=yellow&timeout=3s"
+    http = urllib3.PoolManager(num_pools=2)
+    now = time.monotonic()
+    end = now + wait
+    attempt = 0
+    while now < end:
+        attempt += 1
+        try:
+            resp = http.request("GET", url, timeout=4)
+            if resp.status == 200:
+                LOGGER.debug("elasticsearch is ready")
+                return
+            LOGGER.debug("elasticsearch is not ready %d", attempt)
+        except urllib3.exceptions.MaxRetryError:
+            LOGGER.debug("elasticsearch is not available")
+        time.sleep(3)
+        next = time.monotonic()
+        time.sleep(max(0.5, min((end - next) * 0.7, 5)))
+        now = time.monotonic()
+
+    raise Exception("Elasticsearch is down")
```

Comment:
Are there no other out-of-the-box methods for making sure that ElasticSearch is available? Would it not be possible to implement a standard HEALTHCHECK in the elasticsearch dockerfile?

Reply:
Excellent point, I have just found a nicer way. I'll see if it works.